plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/application/importoperation/GeneralProjectImportOperation.java

```diff
@@ -42,8 +41,9 @@ public GeneralProjectImportOperation(File projectDirectory) {
 
 		IWorkspace workspace = ResourcesPlugin.getWorkspace();
 		IProject project = workspace.getRoot().getProject(getProjectDirectory().getName());
-		overwriteExistingProject(project, workspace, monitor);
-		importToWorkspace(getProjectDirectory(), workspace, monitor);
+		if (overwriteExistingProject(project, workspace, monitor)) {
```

Comment:
Then, if at the first step new files are downloaded and at the second step user declines updating the project description, may we have to explicitly refresh the project in 'else' block? Or it will be refreshed outside of this method?